flink-core/src/main/java/org/apache/flink/util/NetUtils.java

```diff
@@ -498,4 +503,33 @@ public static boolean isValidClientPort(int port) {
     public static boolean isValidHostPort(int port) {
         return 0 <= port && port <= 65535;
     }
+
+    /**
+     * Port wrapper class which holds a {@link FileLock} until it releases. Used to avoid race
+     * condition among multiple threads/processes.
+     */
+    public static class Port implements AutoCloseable {
+        private final int port;
+        private final FileLock fileLock;
+
+        public Port(int port, FileLock fileLock) throws IOException {
+            Preconditions.checkNotNull(fileLock, "FileLock should not be null");
+            Preconditions.checkState(fileLock.isValid(), "FileLock should be locked");
+            this.port = port;
+            this.fileLock = fileLock;
+        }
+
+        public int getPort() {
+            return port;
+        }
+
+        public void release() throws IOException {
+            fileLock.unlockAndDestroy();
+        }
```

Comment:
I'd inline this method into close. It's never directly called.

Reply:
OK. I'll inline it.